src/crypto/polynomials.rs

```diff
@@ -421,6 +421,93 @@ pub fn compute_lagrange_coefficient<C: Ciphersuite>(
     Ok(SerializableScalar(num * den))
 }
 
+/// Computes the Lagrange coefficients (a.k.a. Lagrange basis polynomial)
+/// evaluated at point x in batch.
+/// lamda_i(x) = \prod_j (x - x_j)/(x_i - x_j)  where j != i
+/// Note: if `x` is None then consider it as 0.
+/// Note: `x_j` are elements in `point_set`
+/// Note: `point_set` must not have repeating values.
+pub fn batch_compute_lagrange_coefficients<C: Ciphersuite>(
+    points_set: &[Scalar<C>],
+    x: Option<&Scalar<C>>,
+) -> Result<Vec<SerializableScalar<C>>, ProtocolError> {
+    let n = points_set.len();
+    if n <= 1 {
+        return Err(ProtocolError::InvalidInterpolationArguments);
+    }
+
+    let zero = <C::Group as Group>::Field::zero();
+    let x_val = x.unwrap_or(&zero);
```

Comment:
I'd overload the x variable.
```suggestion
    let x = x.unwrap_or(&zero);
```